apps/kol-amelamdim/context/auth-context-provider.tsx

```diff
@@ -1,22 +1,32 @@
 import React from 'react';
+import axios from '../api';
 
 interface AuthContextTypes {
   isAuthenticated: boolean;
   setAuthenticated: (isAuthenticated: boolean) => void;
+  checkAuthentication: () => Promise<{ success: boolean }>;
 }
 
 export const AuthContext = React.createContext<AuthContextTypes>({
   isAuthenticated: false,
   setAuthenticated: () => null,
+  checkAuthentication: () => null,
 });
 
 export const AuthProvider = ({ children }) => {
   const [isAuthenticated, setAuthenticated] = React.useState<boolean>(false);
+
+  const checkAuthentication = async () => {
+    const { data } = await axios.get('/api/me');
```

Comment:
lets wrap that call with try catch

Reply:
Like we talked, the try catch is in the function that uses `checkAuthentication` (Like we are doing in editorial)